src/data/transform/mod.rs

```diff
@@ -411,21 +429,117 @@ mod tests {
+                raw_input: String::from(""),
+                expect_failure: false,
+                expected_value: Some(json!(null)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'true', expect: true",
+                raw_input: String::from("true"),
+                expect_failure: false,
+                expected_value: Some(json!(true)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'true', type: Boolean, expect: true",
+                raw_input: String::from("true"),
+                expect_failure: false,
+                expected_value: Some(json!(true)),
+                column_type: Some(ColumnType::Boolean),
+            },
+            TestCase {
+                name: "converting: 'false', expect: false",
+                raw_input: String::from("false"),
+                expect_failure: false,
+                expected_value: Some(json!(false)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'false', type: Boolean, expect: false",
+                raw_input: String::from("false"),
+                expect_failure: false,
+                expected_value: Some(json!(false)),
+                column_type: Some(ColumnType::Boolean),
+            },
+            TestCase {
+                name: "converting: '42.42', expect: 42.42",
+                raw_input: String::from("42.42"),
+                expect_failure: false,
+                expected_value: Some(json!(42.42)),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'my string', expect: 'my string'",
+                raw_input: String::from("my string"),
+                expect_failure: false,
+                expected_value: Some(json!("my string")),
+                column_type: None,
+            },
+            TestCase {
+                name: "converting: 'my string', type: String, expect: 'my string'",
+                raw_input: String::from("my string"),
+                expect_failure: false,
+                expected_value: Some(json!("my string")),
+                column_type: Some(ColumnType::String),
+            },
+            TestCase {
+                name: "converting: 'my string', type: Number, expect: Failure",
+                raw_input: String::from("my string"),
+                expect_failure: true,
+                expected_value: None,
+                column_type: Some(ColumnType::Number),
+            },
+            TestCase {
+                name: "converting: 'my string', type: Boolean, expect: Failure",
+                raw_input: String::from("my string"),
+                expect_failure: true,
+                expected_value: None,
+                column_type: Some(ColumnType::Boolean),
+            },
+        ];
+
+        for test_case in test_cases.iter() {
```

Comment:
I think it should also work with the consuming iterator, so maybe you don't need to clone as much
```suggestion
        for test_case in test_cases {
```